src/backend/distributed/commands/utility_hook.c

```diff
@@ -1729,6 +1744,14 @@ RunPreprocessMainDBCommand(Node *parsetree)
 		return;
 	}
 
+
+	if (StatementRequiresUnmarkDistributedFromNonMainDb(parsetree))
+	{
+		List *objectInfoList = GetMarkObjectDistributedParams(parsetree);
+
+		UnMarkObjectDistributedFromNonMainDb(objectInfoList);
+	}
```

Comment:
Should we do that after `EXECUTE_COMMAND_ON_REMOTE_NODES_AS_USER` block so that unmark_object_distributed doesn't complain about that object still exists?